rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleTransactionInstant.java

```diff
@@ -44,18 +49,35 @@ public EthModuleTransactionInstant(RskSystemProperties config, Wallet wallet, Tr
     public synchronized String sendTransaction(Web3.CallArguments args) {
         String txHash = super.sendTransaction(args);
         mineTransaction();
-        return txHash;
+        return getReturnMessage(txHash);
     }
 
     @Override
     public String sendRawTransaction(String rawData) {
         String txHash = super.sendRawTransaction(rawData);
         mineTransaction();
-        return txHash;
+        return getReturnMessage(txHash);
     }
 
     private void mineTransaction() {
         minerServer.buildBlockToMine(blockchain.getBestBlock(), false);
         minerClient.mineBlock();
     }
+
+    /**
+     * When insta-mining we can query the transaction status and return an error response immediately like Ganache.
+     * This does not apply during regular operation because queued transactions are not immediately executed.
+     */
+    private String getReturnMessage(String txHash) {
+        TransactionInfo transactionInfo = blockchain.getTransactionInfo(TypeConverter.stringHexToByteArray(txHash));
+        if (transactionInfo == null) {
+            throw unknownError("Unknown error when sending transaction: transaction wasn't mined");
+        }
+
+        if (!transactionInfo.getReceipt().isSuccessful()) {
+            throw transactionRevertedExecutionError();
```

Comment:
Could we throw the error message from the transaction?
In this example they get the status from debug.transaction https://ethereum.stackexchange.com/questions/6007/how-can-the-transaction-status-from-a-thrown-error-be-detected-when-gas-can-be-e#answer-6011 is there a way to do the same here?

Reply:
I think it's a great idea but the transaction receipt doesn't have any information. Since we don't have it, we would require changes that I believe are out of the scope of this PR.

We do however intend to implement `debug_traceTransaction` in the near future, so maybe it will cover this use case.

Let me know!

Reply:
yeap out of the scope, hope we can add debug_traceTransaction in the future.
Thanks for the reply :)